### ISSUE TITLE
Mention Gitea, and generic git, mecurial, bzr support

### ISSUE THREAD

**Issue by coolaj86** [golang/gddo#578]
As I understand it, the docs just work by git (or mercurial or bzr) and are not tied to specific platforms.

However, the docs on the website are unclear about this:

> GoDoc hosts documentation for Go packages on Bitbucket, GitHub, Google Project Hosting and Launchpad.

I'd suggest updating the copy to read something more similar to this:

> GoDoc hosts documentation for Go packages using git, mercurial, and bzr import URLs.
> This works with popular hosting sites such as Bitbucket, GitHub, Google Project Hosting and Launchpad, as well as peer platforms such as GitLab and Gitea.

Related to https://github.com/golang/gddo/issues/538

**Comment by Bios-Marcel** [comment]
I wonder ... does it support any git provider site? For example I am using https://git.sr.ht , however it can't seem to find this https://godoc.org/?q=git.sr.ht%2F%7Ebiosmarcel%2Fanimaid%2Fanimaid

**Comment by refacto** [comment]
Struggling with the same issue, but with my own `go get`-able site that godoc.org doesn't recognize.
The import path is `import "go.azarus.ch/fuelconv"`

**Comment by arnottcr** [comment]
Looks like the [`pattern`](https://github.com/golang/gddo/blob/master/gosrc/vcs.go#L29) in the [`gddo/gosrc.service`](https://github.com/golang/gddo/blob/master/gosrc/gosrc.go#L158) for generic VCS is insufficient. The `regexp` is pretty strict and only allows `A-Za-z0-9_.\-/`, but [the spec](https://golang.org/ref/spec#Import_declarations) allows for most unicode characters except ```!"#$%&'()*,:;<=>?[\]^`{|}```.


That being said, the two examples you gave are for `git.sr.ht` and `hg.sr.ht`, both of which suffer from `go-get` issues and fail for other reasons:
- https://todo.sr.ht/~sircmpwn/hg.sr.ht/11
- https://todo.sr.ht/~sircmpwn/git.sr.ht/130

Do you have an example `git.sr.ht` repo that I can test with that contains go code at the root? (e.g. `git.sr.ht/~user/pkg`)

**Comment by Bios-Marcel** [comment]
Here: https://git.sr.ht/~biosmarcel/animaid

**Comment by refacto** [comment]
Hi Colin,

Here's an hg repo: https://hg.sr.ht/~azarus/fuelconv

I'll mirror it under https://git.sr.ht/~azarus/fuelconv in a minute.

**Comment by arnottcr** [comment]
yeah, adding the tilde char will unblock these `go get`-able repos:
![image](https://user-images.githubusercontent.com/2526260/54734145-90057d80-4b95-11e9-9234-08871fd0d2f9.png)

I think there is a larger discussion to be had about the long term fix, since this regexp does not follow the spec, but that is for the repo maintainers.

**Comment by dncohen** [comment]
If gddo is going to run a command like `git` or `svn` to fetch the source code, wouldn't it be simpler to just run `go get`?
